src/main/java/com/powsybl/openloadflow/network/LfAction.java

```diff
@@ -335,6 +335,7 @@ private void applyGeneratorChange(LfNetworkParameters networkParameters) {
             double newTargetP = generatorChange.isRelative() ? generator.getTargetP() + generatorChange.activePowerValue() : generatorChange.activePowerValue();
             generator.setTargetP(newTargetP);
             generator.setInitialTargetP(newTargetP);
+            // FIXME should be use generator.reApplyActivePowerControlChecks() instead ?
```

Comment:
@annetill if we do that we break some unit test ? 
Do this behaviour on purpose to not being able to get back the generator to active power control after contingency if target is now correct ?

Reply:
It is an error of the code on main branch, your proposal is the right one and unit tests have to be fixed.